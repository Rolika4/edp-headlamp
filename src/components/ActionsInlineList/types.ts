import { KubeObjectAction } from '../../types/actions';

export interface ActionsInlineListProps {
  actions: KubeObjectAction[];
}
