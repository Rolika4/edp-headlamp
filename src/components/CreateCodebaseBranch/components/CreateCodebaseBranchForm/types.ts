import { EDPCodebaseKubeObjectInterface } from '../../../../k8s/EDPCodebase/types';
import { EDPCodebaseBranchKubeObjectInterface } from '../../../../k8s/EDPCodebaseBranch/types';
import { DeepPartial } from '../../../../types/global';

export interface CreateCodebaseBranchFormProps {
    codebaseData: DeepPartial<EDPCodebaseKubeObjectInterface>;
    editorOpen: boolean;
    setEditorOpen(boolean): void;
    setDialogOpen(boolean): void;
    handleApply(
        newCodebaseBranchData: DeepPartial<EDPCodebaseBranchKubeObjectInterface>,
        newDefaultCodebaseBranchData?: DeepPartial<EDPCodebaseBranchKubeObjectInterface>
    ): void;
    isApplying: boolean;
}
