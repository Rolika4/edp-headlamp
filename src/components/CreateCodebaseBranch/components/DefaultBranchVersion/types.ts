import { CodebaseBranchCreationFormFieldInterface } from '../../types';

export interface DefaultBranchVersionProps extends CodebaseBranchCreationFormFieldInterface {}
