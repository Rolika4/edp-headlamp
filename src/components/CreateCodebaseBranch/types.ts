import { EDPCodebaseKubeObjectInterface } from '../../k8s/EDPCodebase/types';
import { EDPCodebaseBranchKubeObjectInterface } from '../../k8s/EDPCodebaseBranch/types';
import { DeepPartial } from '../../types/global';

export interface CreateCodebaseBranchProps {
    codebaseData: DeepPartial<EDPCodebaseKubeObjectInterface>;
    open: boolean;
    onClose(): void;
    setOpen(boolean): void;
    handleApply(
        newCodebaseBranchData: DeepPartial<EDPCodebaseBranchKubeObjectInterface>,
        newDefaultCodebaseBranchData?: DeepPartial<EDPCodebaseBranchKubeObjectInterface>
    ): void;
    isApplying: boolean;
}
