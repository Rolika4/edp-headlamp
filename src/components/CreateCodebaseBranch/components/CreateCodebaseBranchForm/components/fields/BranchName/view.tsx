import { useFormContext } from 'react-hook-form';
import { MuiCore, React } from '../../../../../../../plugin.globals';
import { FieldEvent } from '../../../../../../../types/forms';
import { FormTextField } from '../../../../../../FormComponents/FormTextField';
import { BranchNameProps } from './types';

const { Grid } = MuiCore;

export const BranchName = ({ names, handleFormFieldChange }: BranchNameProps) => {
    const {
        register,
        control,
        formState: { errors },
    } = useFormContext();

    return (
        <Grid item xs={12}>
            <FormTextField
                {...register(names.branchName.name, {
                    required: `Branch name may contain only: lower-case letters, numbers, slashes, dashes and dots.
                                Can start and end only with lower-case letters and numbers. Minimum 2 characters.
                            `,
                    onBlur: ({ target: { name, value } }: FieldEvent) =>
                        handleFormFieldChange({ name, value }),
                })}
                label={'Branch Name'}
                title={'Type the branch name that will be created in the Version Control System.'}
                placeholder={'Enter Branch Name'}
                control={control}
                errors={errors}
            />
        </Grid>
    );
};
