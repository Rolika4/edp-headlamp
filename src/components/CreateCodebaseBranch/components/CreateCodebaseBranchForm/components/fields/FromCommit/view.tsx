import { useFormContext } from 'react-hook-form';
import { MuiCore, React } from '../../../../../../../plugin.globals';
import { FormTextField } from '../../../../../../FormComponents/FormTextField';
import { FromCommitProps } from './types';

const { Grid } = MuiCore;

export const FromCommit = ({ names, handleFormFieldChange }: FromCommitProps) => {
    const {
        register,
        control,
        formState: { errors },
    } = useFormContext();

    return (
        <Grid item xs={12}>
            <FormTextField
                {...register(names.fromCommit.name, {
                    onBlur: handleFormFieldChange,
                })}
                label={'From Commit Hash '}
                title={
                    'The new branch will be created starting from the selected commit hash. If this field is empty, the Default branch will be used.'
                }
                placeholder={'Enter Commit'}
                control={control}
                errors={errors}
            />
        </Grid>
    );
};
