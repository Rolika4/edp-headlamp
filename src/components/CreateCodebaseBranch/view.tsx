import { ICON_PENCIL } from '../../constants/icons';
import { EDPCodebaseBranchKubeObjectInterface } from '../../k8s/EDPCodebaseBranch/types';
import { Iconify, MuiCore, React, ReactRedux } from '../../plugin.globals';
import { clusterAction } from '../../redux/actions';
import { CreateCodebaseBranchForm } from './components/CreateCodebaseBranchForm';
import { useCreateCodebaseBranch } from './hooks/useCreateCodebaseBranch';
import { useStyles } from './styles';
import { CreateCodebaseBranchProps } from './types';

const { Dialog, DialogContent, Typography, Button } = MuiCore;
const { Icon } = Iconify;
const { useDispatch } = ReactRedux;

export const CreateCodebaseBranch = ({
    codebaseData,
    open,
    setOpen,
    onClose,
}: CreateCodebaseBranchProps): React.ReactElement => {
    const classes = useStyles();
    const [editorOpen, setEditorOpen] = React.useState<boolean>(false);
    const dispatch = useDispatch();

    const { createCodebaseBranch } = useCreateCodebaseBranch(
        () => setOpen(false),
        () => setOpen(true)
    );

    const applyFunc = React.useCallback(
        async (
            newCodebaseBranchData: EDPCodebaseBranchKubeObjectInterface
        ): Promise<EDPCodebaseBranchKubeObjectInterface | undefined> =>
            createCodebaseBranch(newCodebaseBranchData),
        [createCodebaseBranch]
    );
    const handleApply = React.useCallback(
        async (newCodebaseBranchData: EDPCodebaseBranchKubeObjectInterface): Promise<void> => {
            const cancelUrl = location.pathname;
            const {
                spec: { branchName },
            } = newCodebaseBranchData;

            dispatch(
                clusterAction(() => applyFunc(newCodebaseBranchData), {
                    startMessage: `Applying ${branchName}`,
                    cancelledMessage: `Cancelled applying ${branchName}`,
                    successMessage: `Applied ${branchName}`,
                    errorMessage: `Failed to apply ${branchName}`,
                    cancelUrl,
                })
            );
        },
        [applyFunc, dispatch]
    );

    return (
        <Dialog open={open} onClose={onClose} fullWidth className={classes.dialogRoot}>
            <div className={classes.dialog} data-testid={'create-codebase-branch'}>
                <div className={classes.dialogTitle}>
                    <Typography variant={'h5'}>{`Create new branch`}</Typography>
                    <Button
                        startIcon={<Icon icon={ICON_PENCIL} />}
                        size="small"
                        component={'button'}
                        onClick={() => setEditorOpen(true)}
                    >
                        Edit YAML
                    </Button>
                </div>
                <DialogContent className={classes.dialogContent}>
                    <CreateCodebaseBranchForm
                        codebaseData={codebaseData}
                        editorOpen={editorOpen}
                        setEditorOpen={setEditorOpen}
                        handleApply={handleApply}
                        setDialogOpen={setOpen}
                    />
                </DialogContent>
            </div>
        </Dialog>
    );
};
