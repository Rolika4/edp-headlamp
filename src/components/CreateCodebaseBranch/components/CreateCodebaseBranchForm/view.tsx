import type { DialogProps } from '@material-ui/core/Dialog';
import lodashOmit from 'lodash.omit';
import { FormProvider, useForm } from 'react-hook-form';
import { useHandleEditorSave } from '../../../../hooks/useHandleEditorSave';
import { EDPCodebaseBranchKubeObjectInterface } from '../../../../k8s/EDPCodebaseBranch/types';
import { MuiCore, pluginLib, React } from '../../../../plugin.globals';
import { DeepPartial } from '../../../../types/global';
import { Render } from '../../../Render';
import { BranchName } from './components/fields';
import { FromCommit } from './components/fields/FromCommit';
import { useDefaultValues } from './hooks/useDefaultValues';
import { useEditorCode } from './hooks/useEditorCode';
import { CODEBASE_BRANCH_BACKWARDS_NAME_MAPPING, CODEBASE_BRANCH_NAMES } from './names';
import { useStyles } from './styles';
import { CreateCodebaseBranchFormProps } from './types';

const { Button, Grid } = MuiCore;

const {
    CommonComponents: { EditorDialog },
} = pluginLib;

export const CreateCodebaseBranchForm = ({
    codebaseData,
    editorOpen,
    setEditorOpen,
    handleApply,
    setDialogOpen,
}: CreateCodebaseBranchFormProps): React.ReactElement => {
    const classes = useStyles();

    const { baseDefaultValues } = useDefaultValues({ names: CODEBASE_BRANCH_NAMES });

    const [formValues, setFormValues] =
        React.useState<DeepPartial<EDPCodebaseBranchKubeObjectInterface>>(baseDefaultValues);

    const methods = useForm();

    const {
        handleSubmit,
        reset,
        resetField,
        formState: { isDirty },
        setValue,
    } = methods;

    const handleFormFieldChange = React.useCallback(({ target: { name, value } }) => {
        setFormValues(prev => {
            if (Object.hasOwn(CODEBASE_BRANCH_NAMES[name], 'notUsedInFormData')) {
                return prev;
            }

            if (value === undefined) {
                return lodashOmit(prev, name);
            }

            return {
                ...prev,
                [name]: value,
            };
        });
    }, []);

    const handleResetFields = React.useCallback(() => {
        setFormValues(baseDefaultValues);
        reset();
    }, [baseDefaultValues, reset]);

    const { handleEditorSave } = useHandleEditorSave({
        names: CODEBASE_BRANCH_NAMES,
        backwardNames: CODEBASE_BRANCH_BACKWARDS_NAME_MAPPING,
        setValue,
        handleFormFieldChange,
        formValues,
        resetField,
    });

    const { editorCode } = useEditorCode({
        names: CODEBASE_BRANCH_NAMES,
        formValues,
        codebaseName: codebaseData.metadata.name,
        namespace: codebaseData.metadata.namespace,
    });

    const onEditorSave = React.useCallback(
        (editorPropsObject: EDPCodebaseBranchKubeObjectInterface) => {
            handleEditorSave(editorPropsObject);
            setEditorOpen(false);
        },
        [handleEditorSave, setEditorOpen]
    );

    const muDialogProps: DialogProps = {
        open: editorOpen,
    };

    const onSubmit = React.useCallback(() => {
        handleApply(editorCode);
    }, [editorCode, handleApply]);

    return (
        <FormProvider {...methods}>
            <div className={classes.form}>
                <form onSubmit={handleSubmit(onSubmit)}>
                    <div className={classes.formInner}>
                        <Grid container spacing={2}>
                            <Grid item xs={12}>
                                <BranchName
                                    names={CODEBASE_BRANCH_NAMES}
                                    handleFormFieldChange={handleFormFieldChange}
                                />
                            </Grid>
                            <Grid item xs={12}>
                                <FromCommit
                                    names={CODEBASE_BRANCH_NAMES}
                                    handleFormFieldChange={handleFormFieldChange}
                                />
                            </Grid>
                        </Grid>
                    </div>
                    <div className={classes.actions}>
                        <Button
                            onClick={handleResetFields}
                            size="small"
                            component={'button'}
                            disabled={!isDirty}
                        >
                            undo changes
                        </Button>
                        <Button
                            onClick={() => setDialogOpen(false)}
                            size="small"
                            component={'button'}
                            style={{ marginLeft: 'auto' }}
                        >
                            cancel
                        </Button>
                        <Button
                            type={'submit'}
                            variant={'contained'}
                            color={'primary'}
                            size="small"
                            disabled={!isDirty}
                        >
                            apply
                        </Button>
                    </div>
                </form>
            </div>
            <Render condition={!!editorOpen}>
                <EditorDialog
                    {...muDialogProps}
                    item={editorCode}
                    onClose={() => setEditorOpen(false)}
                    onSave={onEditorSave}
                />
            </Render>
        </FormProvider>
    );
};
