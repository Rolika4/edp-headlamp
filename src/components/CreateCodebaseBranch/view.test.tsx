/**
 * @jest-environment jsdom
 */

import { describe } from '@jest/globals';
import { configureStore } from '@reduxjs/toolkit';
import { render, screen } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';
import React from 'react';
import { Provider } from 'react-redux';
import { CreateCodebaseBranchProps } from './types';
import { CreateCodebaseBranch } from './view';

describe('CreateCodebaseBranch', () => {
    it('should render correctly', () => {
        const props: CreateCodebaseBranchProps = {
            codebaseData: {
                apiVersion: 'apiVersion',
                kind: 'CodebaseBranch',
                metadata: {
                    name: 'test-app-name',
                    namespace: 'test-namespace',
                },
            },
            open: true,
            setOpen: () => {},
            onClose: () => {},
        };
        const store = configureStore({
            reducer: () => ({}),
        });

        render(
            <Provider store={store}>
                <SnackbarProvider>
                    <CreateCodebaseBranch {...props} />
                </SnackbarProvider>
            </Provider>
        );

        expect(screen.getByTestId('create-codebase-branch')).toMatchSnapshot();
    });
});
