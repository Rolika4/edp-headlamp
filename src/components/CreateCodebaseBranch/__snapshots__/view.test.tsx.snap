// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCodebaseBranch should render correctly 1`] = `
<div
  class="makeStyles-dialog-1"
  data-testid="create-codebase-branch"
>
  <div
    class="makeStyles-dialogTitle-2"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      Create new branch
    </h5>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall"
        >
          <span />
        </span>
        Edit YAML
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDialogContent-root makeStyles-dialogContent-3"
  >
    <div
      class="makeStyles-form-4"
    >
      <form>
        <div
          class="makeStyles-formInner-5"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiFormControl-root"
                  >
                    <label
                      class="MuiFormControlLabel-root"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <input
                            class="PrivateSwitchBase-input-10"
                            data-indeterminate="false"
                            name="release"
                            type="checkbox"
                            value="false"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                      >
                        <span
                          style="display: inline-block; margin-top: 0.125rem;"
                        >
                          <span
                            class="makeStyles-labelWrap-11"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-12 MuiTypography-body1"
                            >
                              Release branch
                            </span>
                          </span>
                        </span>
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <span
                          class="makeStyles-labelWrap-11"
                        >
                          <span
                            class="MuiTypography-root makeStyles-label-12 MuiTypography-body1"
                          >
                            Branch Name
                          </span>
                          <span />
                        </span>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth"
                        >
                          <div
                            class="MuiFormControl-root MuiTextField-root"
                          >
                            <div
                              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                            >
                              <input
                                aria-invalid="false"
                                class="MuiInputBase-input MuiInput-input"
                                name="branchName"
                                placeholder="Enter Branch Name"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <span
                          class="makeStyles-labelWrap-11"
                        >
                          <span
                            class="MuiTypography-root makeStyles-label-12 MuiTypography-body1"
                          >
                            From Commit Hash 
                          </span>
                          <span />
                        </span>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      >
                        <div
                          class="MuiFormControl-root MuiFormControl-fullWidth"
                        >
                          <div
                            class="MuiFormControl-root MuiTextField-root"
                          >
                            <div
                              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                            >
                              <input
                                aria-invalid="false"
                                class="MuiInputBase-input MuiInput-input"
                                name="fromCommit"
                                placeholder="Enter Commit"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-11"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-12 MuiTypography-body1"
                            >
                              Branch version
                            </span>
                            <span />
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiFormControl-root MuiTextField-root"
                            >
                              <div
                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              >
                                <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiInput-input"
                                  name="branchVersionStart"
                                  placeholder="0.0.0"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
                  style="display: flex; justify-content: flex-end; flex-direction: column;"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiFormControl-root MuiTextField-root"
                            >
                              <div
                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              >
                                <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiInput-input"
                                  name="branchVersionPostfix"
                                  placeholder="SNAPSHOT"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-actions-6"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              undo changes
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            style="margin-left: auto;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              apply
            </span>
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
